ros/processor/inventory_events_consumer.py

```diff
@@ -103,27 +108,79 @@ def host_create_update_events(self, msg):
         """ Process created/updated message ( create system record, store new report )"""
         self.prefix = "PROCESSING Create/Update EVENT"
         with app.app_context():
-            print(msg)
-            self.process_system_details(msg)
+            if 'is_ros' in msg['platform_metadata']:
+                self.process_system_details(msg)
 
     def process_system_details(self, msg):
         """ Store new system information (stale, stale_warning timestamp) and return internal DB id"""
         host = msg['host']
+        insights_report_tar = self._download_report(msg['platform_metadata']['url'])
+        performance_record = self._extract_performance_record(insights_report_tar)
+        if performance_record:
+            performance_score = self._calculate_performance_score(performance_record)
 
-        account = get_or_create(
-            db.session, RhAccount, 'account',
-            account=host['account']
-        )
-
-        system = get_or_create(
-            db.session, System, 'inventory_id',
-            account_id=account.id,
-            inventory_id=host['id'],
-            display_name=host['display_name'],
-            fqdn=host['fqdn'],
-            cloud_provider=host['system_profile']['cloud_provider']
-        )
-        # Commit changes
-        db.session.commit()
-        LOG.warning("Refreshed system %s (%s) belonging to account: %s (%s)",
-                    system.inventory_id, system.id, account.account, account.id)
+            account = get_or_create(
+                db.session, RhAccount, 'account',
+                account=host['account']
+            )
+
+            system = get_or_create(
+                db.session, System, 'inventory_id',
+                account_id=account.id,
+                inventory_id=host['id'],
+                display_name=host['display_name'],
+                fqdn=host['fqdn'],
+                cloud_provider=host['system_profile']['cloud_provider']
+            )
+
+            get_or_create(
+                db.session, PerformanceProfile, ['system_id', 'report_date'],
+                system_id=system.id,
+                performance_record=performance_record,
+                performance_score=performance_score,
+                report_date=date.today()
+            )
+
+            # Commit changes
+            db.session.commit()
+            LOG.warning("Refreshed system %s (%s) belonging to account: %s (%s)",
+                        system.inventory_id, system.id, account.account, account.id)
+
+    def _download_report(self, report_url):
+        download_response = requests.get(report_url)
+        if download_response.status_code != HTTPStatus.OK:
+            print("Unable to download the report")
+        return download_response.content
+
+    def _extract_performance_record(self, report_tar):
+        tar = tarfile.open(fileobj=BytesIO(report_tar), mode='r:*')
+        files = tar.getmembers()
```

Comment:
Instead of `getmembers` and below iteration on files, will it be possible to make use of `getmember` and fetch whichever required?
https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.getmember

Reply:
Using `getmember` will be bit hard as tar files contain the hostname of the client system, So for using `getmember` first we will need to find out the hostname of the system. Also, I not sure if tar will always be having hostname in the file path. 
```
>>> tar.getmember('./insights-ip-172-31-0-61.ap-south-1.compute.internal-20210212124759/data/insights_commands/pmlogsummary')
<TarInfo './insights-ip-172-31-0-61.ap-south-1.compute.internal-20210212124759/data/insights_commands/pmlogsummary' at 0x7f67e41cf1c0>
```
Let keep this for now, we can fix it later as a scope of performance improvement.